Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.CardBrowser}: com.ichi2.anki.exception.UserSubmittedException: Unable to instantiate fragment X3.h: could not find Fragment constructor

https://ankidroid.org/acra/app/1/bug/253877/report/5a4b008d-427b-4876-a085-0095795f52b8

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.CardBrowser}: com.ichi2.anki.exception.UserSubmittedException: Unable to instantiate fragment X3.h: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4014)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4167)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:99)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2628)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:222)
	at android.os.Looper.loop(Looper.java:314)
	at android.app.ActivityThread.main(ActivityThread.java:8661)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:565)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1081)
Caused by: com.ichi2.anki.exception.UserSubmittedException: Unable to instantiate fragment X3.h: could not find Fragment constructor
	at androidx.fragment.app.Fragment$InstantiationException.<init>(Fragment.java:577)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:692)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:663)
	at com.ichi2.utils.ExtendedFragmentFactory.instantiate(ExtendedFragmentFactory.kt:50)
	at com.ichi2.anki.dialogs.tags.TagsDialogFactory.instantiate(TagsDialogFactory.kt:27)
	at com.ichi2.utils.ExtendedFragmentFactory.instantiate(ExtendedFragmentFactory.kt:50)
	at com.ichi2.anki.export.ExportDialogsFactory.instantiate(ExportDialogsFactory.java:32)
	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:87)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:92)
	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2846)
	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:3005)
	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117)
	at androidx.fragment.app.FragmentActivity.lambda$init$3(FragmentActivity.java:139)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.kt:84)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.kt:331)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:216)
	at com.ichi2.anki.AnkiActivity.onCreate(AnkiActivity.kt:114)
	at com.ichi2.anki.CardBrowser.onCreate(CardBrowser.kt:367)
	at android.app.Activity.performCreate(Activity.java:8886)
	at android.app.Activity.performCreate(Activity.java:8851)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1467)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3996)
	... 12 more
Caused by: java.lang.NoSuchMethodException: X3.h.<init> []
	at java.lang.Class.getConstructor0(Class.java:3395)
	at java.lang.Class.getConstructor(Class.java:2487)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:677)
	... 29 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt
@@ -1562,7 +1562,7 @@ open class CardBrowser :
     }
 
     private fun showOptionsDialog() {
-        val dialog = BrowserOptionsDialog(viewModel.cardsOrNotes, viewModel.isTruncated)
+        val dialog = BrowserOptionsDialog.newInstance(viewModel.cardsOrNotes, viewModel.isTruncated)
         dialog.show(supportFragmentManager, "browserOptionsDialog")
     }
 

--- a/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/BrowserOptionsDialog.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/BrowserOptionsDialog.kt
@@ -26,6 +26,7 @@ import android.widget.RadioButton
 import android.widget.RadioGroup
 import androidx.annotation.IdRes
 import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.core.os.bundleOf
 import androidx.fragment.app.activityViewModels
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import com.ichi2.anki.R
@@ -33,14 +34,16 @@ import com.ichi2.anki.browser.CardBrowserViewModel
 import com.ichi2.anki.model.CardsOrNotes
 import timber.log.Timber
 
-class BrowserOptionsDialog(private val cardsOrNotes: CardsOrNotes, private val isTruncated: Boolean) : AppCompatDialogFragment() {
+class BrowserOptionsDialog : AppCompatDialogFragment() {
     private lateinit var dialogView: View
 
     private val viewModel: CardBrowserViewModel by activityViewModels()
 
     private val positiveButtonClick = { _: DialogInterface, _: Int ->
-        @IdRes val selectedButtonId = dialogView.findViewById<RadioGroup>(R.id.select_browser_mode).checkedRadioButtonId
-        val newCardsOrNotes = if (selectedButtonId == R.id.select_cards_mode) CardsOrNotes.CARDS else CardsOrNotes.NOTES
+        @IdRes val selectedButtonId =
+            dialogView.findViewById<RadioGroup>(R.id.select_browser_mode).checkedRadioButtonId
+        val newCardsOrNotes =
+            if (selectedButtonId == R.id.select_cards_mode) CardsOrNotes.CARDS else CardsOrNotes.NOTES
         if (cardsOrNotes != newCardsOrNotes) {
             viewModel.setCardsOrNotes(newCardsOrNotes)
         }
@@ -48,6 +51,25 @@ class BrowserOptionsDialog(private val cardsOrNotes: CardsOrNotes, private val i
 
         if (newTruncate != isTruncated) {
             viewModel.setTruncated(newTruncate)
+        }
+    }
+
+    private val cardsOrNotes by lazy {
+        when (arguments?.getBoolean(CARDS_OR_NOTES_KEY)) {
+            true -> CardsOrNotes.CARDS
+            false -> CardsOrNotes.NOTES
+            null -> {
+                // Default case, and what we'll do if there were no arguments supplied
+                Timber.w("BrowserOptionsDialog instantiated without configuration.")
+                CardsOrNotes.CARDS
+            }
+        }
+    }
+
+    private val isTruncated by lazy {
+        arguments?.getBoolean(IS_TRUNCATED_KEY) ?: run {
+            Timber.w("BrowserOptionsDialog instantiated without configuration.")
+            false
         }
     }
 
@@ -78,6 +100,21 @@ class BrowserOptionsDialog(private val cardsOrNotes: CardsOrNotes, private val i
             }
             this.setPositiveButton(getString(R.string.dialog_ok), DialogInterface.OnClickListener(function = positiveButtonClick))
             this.create()
+        }
+    }
+
+    companion object {
+        private const val CARDS_OR_NOTES_KEY = "cardsOrNotes"
+        private const val IS_TRUNCATED_KEY = "isTruncated"
+
+        fun newInstance(cardsOrNotes: CardsOrNotes, isTruncated: Boolean): BrowserOptionsDialog {
+            Timber.i("BrowserOptionsDialog::newInstance")
+            return BrowserOptionsDialog().apply {
+                arguments = bundleOf(
+                    CARDS_OR_NOTES_KEY to (cardsOrNotes == CardsOrNotes.CARDS),
+                    IS_TRUNCATED_KEY to isTruncated
+                )
+            }
         }
     }
 }

